src/compiler/checker.ts

```diff
@@ -16729,6 +16743,24 @@ namespace ts {
             return true;
         }
 
+        function checkAndAggregateParameterExpressionTypes(parameter: ParameterDeclaration): Type[] {
+            const func = <FunctionLikeDeclaration>parameter.parent;
+            const usageTypes: Type[] = [];
+            forEachInvocation(<Block>func.body, invocation => {

```

Comment:
Clearly, only collecting invocation isn't enough. But I wonder how other expressions can be handled. For example, how `arg.assigment = 123` is handled.